### ISSUE TITLE
Accessibility: macOS `Problem` tooltip does not support opening an accessible view

### ISSUE THREAD

**Issue by jquesada2016** [microsoft/vscode#263863]
Type: <b>Bug</b>

On macOS using VoiceOver, if I am in a file that has an error, when I press `F8`, it sometimes is read aloud, sometimes not. But, if I try to open an accessible view to read the error message, nothing happens.

To reprodoce, first open a file that has an error reported by VS Code, where `F8` would jump to it and open a tooltip. Then, press `Option + F2`. Notice that no accessible view opens.

The accessible view seems to open everywhere else, except for problems, making issues unreadable.

VS Code version: Code 1.103.2 (Universal) (6f17636121051a53c88d3e605c491d22af2ba755, 2025-08-20T16:45:34.255Z)
OS version: Darwin arm64 24.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M4 Max (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: enabled_on<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 3, 2|
|Memory (System)|128.00GB (71.68GB free)|
|Process Argv|--crash-reporter-id 9eef8a64-0777-4b98-8906-d2ab725c36fc|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (113)</summary>

Extension|Author (truncated)|Version
---|---|---
language-x86-64-assembly|13x|3.1.5
svelte-bundle|1Yi|1.0.0
monokai-charcoal-high-contrast|74t|3.5.0
ng-template|Ang|20.2.2
slidev|ant|51.4.0
svelte-intellisense|ard|0.7.1
flutter-redux-gen|Bal|3.0.9
calva|bet|2.0.524
calva-spritz|bet|1.0.5
markdown-mermaid|bie|1.28.0
vscode-tailwindcss|bra|0.14.26
better-toml|bun|0.3.2
rescript-vscode|che|1.64.0
vscode-postgres|cko|1.4.3
doxdocgen|csc|1.4.0
dart-code|Dar|3.116.0
flutter|Dar|3.116.0
vscode-deno|den|3.45.2
dhall-lang|dha|0.0.4
vscode-dhall-lsp-server|dha|0.0.4
docker|doc|0.16.0
vscode-wasm|dts|1.4.1
rust-syntax|dus|0.6.1
vscode-hyperscript-org|dz4|0.1.5
gitlens|eam|17.4.1
elm-ls-vscode|elm|2.8.0
prettier-vscode|esb|11.0.0
syntax-highlighter|evg|0.5.0
vscode-svelte-snippets|fiv|0.5.0
remotehub|Git|0.64.0
vscode-github-actions|git|0.27.2
gleam|gle|2.12.1
gc-excelviewer|Gra|4.2.64
vscode-test-explorer|hbe|2.22.1
vscode-drawio|hed|1.9.0
mysql-syntax|jak|1.3.1
better-cpp-syntax|jef|1.27.1
ghdl-interface|joh|1.1.2
cmake-language-support-vscode|jos|0.0.9
rust-doc-viewer|JSc|4.2.0
edgedb|mag|0.2.3
nc-gcode|ML|0.13.0
vscode-scss|mrm|0.10.0
vscode-containers|ms-|2.1.0
vscode-docker|ms-|2.0.0
vscode-dotnet-runtime|ms-|2.3.7
playwright|ms-|1.1.15
debugpy|ms-|2025.10.0
isort|ms-|2025.0.0
python|ms-|2025.12.0
vscode-pylance|ms-|2025.7.1
vscode-python-envs|ms-|1.2.0
jupyter|ms-|2025.7.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.3.0
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.422.1
remote-ssh|ms-|0.120.0
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.99.0
azure-repos|ms-|0.40.0
cmake-tools|ms-|1.21.36
cpptools|ms-|1.26.3
cpptools-extension-pack|ms-|1.3.1
powershell|ms-|2025.2.0
remote-explorer|ms-|0.5.0
remote-repositories|ms-|0.42.0
test-adapter-converter|ms-|0.2.1
vscode-serial-monitor|ms-|0.13.1
vsliveshare|ms-|1.0.5959
veriloghdl|msh|1.16.1
sqltools|mtx|0.28.4
vscode-purty|mva|0.6.0
gradle-language|nac|0.2.3
vscode-python-typehint|njq|1.5.1
ide-purescript|nwo|0.26.6
language-purescript|nwo|0.2.10
rust-and-friends|nyx|1.0.0
ocaml-platform|oca|1.32.0
fluent-syntax-highlighting|pau|0.0.1
material-icon-theme|PKi|5.26.0
rust-targets|Pol|1.2.1
awesome-vhdl|puo|0.0.1
vscode-thunder-client|ran|2.37.5
java|red|1.44.0
vscode-xml|red|0.29.0
vscode-yaml|red|1.18.0
vscode-gradle-extension-pack|ric|0.0.4
vscode-modern-vhdl-support|rjy|1.0.6
rust-analyzer|rus|0.3.2593
crates|ser|0.6.7
shader|sle|1.1.5
surrealql|sur|0.2.16
svelte-vscode|sve|109.11.0
even-better-toml|tam|0.21.2
tauri-vscode|tau|0.2.9
vscode-nushell-lang|The|1.10.0
vsfire|tob|1.4.1
cmake|twx|0.0.17
vscode-lldb|vad|1.11.4
intellicode-api-usage-examples|Vis|0.2.9
vscodeintellicode|Vis|1.3.2
vscode-gradle|vsc|3.16.4
vscode-java-debug|vsc|0.58.2
vscode-java-dependency|vsc|0.24.1
vscode-java-pack|vsc|0.29.2
vscode-java-test|vsc|0.43.1
vscode-maven|vsc|0.44.0
volar|Vue|3.0.6
wgsl-analyzer|wgs|0.11.39
jinja|who|0.0.8
rust-mod-generator|Zha|1.0.12

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
pythonvspyt551cf:31249601
binariesv615:30325510
nativeloc1:31344060
dwcopilot:31170013
dwoutputs:31242946
copilot_t_ci:31333650
e5gg6876:31282496
pythoneinst12:31285622
996jf627:31283433
pythonrdcb7:31342333
usemplatestapi:31297334
747dc170:31275177
aj953862:31281341
9d2cg352:31346308
nesew2to5:31336538
agentclaude:31374413
4f60g487:31327383
nes-set-on:31340697
testaa123cf:31335227
6abeh943:31336334
0927b901:31350571
ji9b5146:31348712
custommodel2t:31371781
45650338:31358607
0cj2b977:31352657
justeven_python_cf:31371805
gaj49834:31362110
pylancealldocst:31371774
gemagent1:31368469
cloudbuttont:31374192
retryenabled:31370050
3efgi100_wstrepl:31374188

```

</details>

**Comment by meganrogge** [comment]
Hi, if there's info missing from the problems view, pls let us know. We don't have an accessible view for this and there shouldn't need to be one. We should make the feature itself accessible.

**Comment by jquesada2016** [comment]
I didn't fully understand you're response.

I interpreted one of two things.

Whenever a hover tooltip pops up (i.e. by pressing `F8`), then this tooltip should not have an accessible view?

Or my other interpretation:

Whenever I use the `Problems` tab, which is normally grouped with the `Terminal`, `Output`, and other such tabs, should not require an accessible view?

Regarding the former, if I have the tooltip up, sometimes VoiceOver bugs out and doesn't read aloud the error, and when it does read the error, it doesn't bring it up in my braille display, so I'm unable to read it. If this tooltip had an accessible view, then I would be able to read it, especially when the error has more than a single line, often the case when I get a complex type error like with `TypeScript`, `PureScript`, or `Rust` type errors.

If it's the latter, my issue with using the `Problems` tab is that as I navigate through the errors and messages, the IDE jumps around in my code, and I often lose the place I was, taking me a few seconds to get my bearing.

My normal workflow for fixing errors is:

I peruse through the line before and after the error to get a bit of context. Once I have that context, I move up and down until I hear the "error" sound. Then I move forward through the line until I hear the "error" sound again, which tells me I'm in the part of the underlined error. At this point, I press `Option + F8` to bring up the error tooltip, which is where I struggle to read it with VoiceOver.

My current workaround is to use the little vision I have to see the hover tooltip, and use my mouse to hover over it, at which point macOS's other accessibility feature kicks in and reads the content aloud to me.

**Comment by meganrogge** [comment]
Could you pls create a screen recording of what you're describing? 

I'm having trouble understanding. Thank you

**Comment by jquesada2016** [comment]
Of course, here's a video. Please note that in this case, I'm using Rust and I'm showing it with a warning, but the same thing happens if it's an error, as well as another language.

**Comment by jquesada2016** [comment]
Not sure what's going on but GitHub isn't letting me upload the video. Please let me know how I can share it with you.

**Comment by meganrogge** [comment]
Pls upload it to github

**Comment by jquesada2016** [comment]
I was unable to upload the video to GitHub, so I uploaded it to YouTube as an unlisted video. You can watch the video [here](https://youtu.be/TYuTEwTyzxc).

**Comment by meganrogge** [comment]
Great explanation 👏🏼 thank you. I totally see what you mean. I think we can add this.

**Comment by meganrogge** [comment]
@jooyoungseo, is this something you'd also be interested in?
